strax/processing/hitlets.py

```diff
@@ -0,0 +1,526 @@
+    # _concat_overlapping_hits numba crashes.
+    last_hit_in_channel = np.zeros(nchannels,
+                                   dtype=(hit_dtype
+                                          + [(('End time of the interval (ns since unix epoch)',
+                                               'endtime'), np.int64)]))
+
+    if len(hits):
+        hits = _concat_overlapping_hits(hits, extensions, first_channel, last_hit_in_channel)
+    return hits
+
+
+@utils.growing_result(strax.hit_dtype, chunk_size=int(1e4))
+@numba.njit(nogil=True, cache=True)
+def _concat_overlapping_hits(hits,
+                             extensions,
+                             first_channel,
+                             last_hit_in_channel,
+                             _result_buffer=None):
+    buffer = _result_buffer
+    offset = 0
+
+    le, re = extensions
+    dt = hits['dt'][0]
+    assert np.all(hits['dt'] == dt), 'All hits must have the same dt!'
+
+    for h in hits:
+        st = h['time'] - int(le * h['dt'])
+        et = strax.endtime(h) + int(re * h['dt'])
+        hc = h['channel']
+        r_i = h['record_i']
+
+        lhc = last_hit_in_channel[hc - first_channel]
+        # Have not found any hit in this channel yet:
+        if lhc['time'] == 0:
+            lhc['time'] = st
+            lhc['endtime'] = et
+            lhc['channel'] = hc
+            lhc['record_i'] = h['record_i']
+            lhc['dt'] = dt
+
+        # Checking if events overlap:
+        else:
+            if lhc['endtime'] >= st:
+                # Yes, so we have to update only the end_time:
+                lhc['endtime'] = et
+            else:
+                # No, this means we have to save the previous data and update lhc:
+                res = buffer[offset]
+                res['time'] = lhc['time']
+                res['length'] = (lhc['endtime'] - lhc['time'])//lhc['dt']
+                res['channel'] = lhc['channel']
+                res['record_i'] = lhc['record_i']
+                res['dt'] = lhc['dt']
+                offset += 1
+                if offset == len(buffer):
+                    yield offset
+                    offset = 0
+
+                # Updating current last hit:
+                lhc['time'] = st
+                lhc['endtime'] = et
+                lhc['channel'] = hc
+                lhc['record_i'] = r_i
+
+    # We went through so now we have to save all remaining hits:
+    mask = last_hit_in_channel['time'] != 0
+    for lhc in last_hit_in_channel[mask]:
+        res = buffer[offset]
+        res['time'] = lhc['time']
+        res['channel'] = lhc['channel']
+        res['length'] = (lhc['endtime'] - lhc['time'])//lhc['dt']
+        res['record_i'] = lhc['record_i']
+        res['dt'] = lhc['dt']
+        offset += 1
+        if offset == len(buffer):
+            yield offset
+            offset = 0
+    yield offset
+
+
+@export
+@numba.njit(nogil=True, cache=True)
+def refresh_hit_to_hitlets(hits, hitlets):
```

Comment:
I am not sure if you can ask within numba if a specific numpy field exists. At least .dtype.names etc. is not working, but I can check.

Reply:
can you also not pass on the dtype names as a typing.List?

Reply:
I never worked with numba lists, but I can give it a try.

Reply:
a numpy array of strings should also work I expect it was more about the wapping in a non-numba function to extract the dtypes

Reply:
I just remembered yesterday, that I think tied this already.  Since I was not sure I just gave it another shot. No it does not work. My guess would be that under the hood of numba it does not work for the same reason as array.dtype.names does not.